client/my-sites/plans/current-plan/controller.jsx

```diff
@@ -20,7 +20,9 @@ export function currentPlan( context, next ) {
 	const isEcommerceTrial = currentPlanSlug === PLAN_ECOMMERCE_TRIAL_MONTHLY;
 	const isEntrepreneurTrial = isEcommerceTrial && ! purchase?.isWooExpressTrial;
 
-	if ( isFreePlan || isEntrepreneurTrial ) {
+	// When redirecting from the checkout page, selectedSite's plan hasn't had time to update yet, use success param to
+	// to bypass the isFreePlan check here and allow loading of /plans/my-plan.
+	if ( ( ! context.query.hasOwnProperty( 'success' ) && isFreePlan ) || isEntrepreneurTrial ) {
 		page.redirect( `/plans/${ selectedSite.slug }` );
 
 		return null;
```

Comment:
Might remove this and go back to /plans, /plans/my-plan feels out of date and the same info is on the plans page now too.

Reply:
I think we can remove this now.